server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1320,10 +1320,11 @@ public Operation(Term uid, long seqNo, long primaryTerm, long version, VersionTy
             REPLICA,
             PEER_RECOVERY,
             LOCAL_TRANSLOG_RECOVERY,
-            LOCAL_RESET;
+            LOCAL_RESET,
+            SNAPSHOT_RECOVERY;
 
             public boolean isRecovery() {
-                return this == PEER_RECOVERY || this == LOCAL_TRANSLOG_RECOVERY;
+                return this == PEER_RECOVERY || this == LOCAL_TRANSLOG_RECOVERY || this == SNAPSHOT_RECOVERY;
```

Comment:
I looked at other usages of these constants and noticed that `SNAPSHOT_RECOVERY` is not covered now in `InternalEngine.canOptimizeAddDocument`. Will we fail with append-only indexing ops?